Automattic/jetpack
# Search: Add DB post type breakdown endpoint

Fixes #26459 

#### Changes proposed in this Pull Request:
The PR proposes to add a `/jetpack/v4/search/local-stats` endpoint for WPCOM to use for record number estimation. The current solution uses `/wp/v2/search` which is not consistent with all the other places.

The PR adds a method `Automattic\Jetpack\My_Jetpack\Products\Search_Stats::get_post_type_breakdown()` to generate post type breakdown for the site, and also refactors `Automattic\Jetpack\My_Jetpack\Products\Search_Stats::estimate_count()` to use it, just to be consistent with everywhere else.


#### Other information:


#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Ensure all PHP tests pass
* Navigate to a site with Search plugin and valid Search subscription
* Open `/wp-admin/admin.php?page=jetpack-search` and inspect network
* Ensure `estimate_record_count` has the indexable number of posts on the site

![image](https://user-images.githubusercontent.com/1425433/192679068-149168e6-8cfc-4f9e-9845-1340fcb01f5d.png)
